packages/react-server/core/logging/response.js

```diff
@@ -0,0 +1,86 @@
+var  SuperLogger = require('winston').Transport
+,	         RLS = require('../util/RequestLocalStorage').getNamespace();
+
+// A subset of stats that are logged are not associated with requests
+// or occur before the request context is initialized. Simply ignore
+// those logs here.
+var queue = () => {
+	if (RLS.isActive()) {
+		return RLS().queue || (RLS().queue = []);
+	}
+	else {
+		return [];
+	}
+}
+
+var pushToQueue = (tuple) => {
+	if (RLS.isActive() && !!RLS().doLog) {
+		queue().push(tuple);
+	}
+}
+class ResponseLogger extends SuperLogger {
+	constructor(options) {
+		super();
+		this.name = 'ResponseLogger';
+		this.level = options.level || 'debug';
+		this.module = options.name;
+		this.lastModuleToken  = options.name.split('.').pop();
+	}
+
+	log(level, msg, meta, callback) {
+		var tuple = [
+			this.module,
+			msg,
+			meta[this.key],
+			this.lastModuleToken,
+		];
+		pushToQueue(tuple);
```

Comment:
Let's make this even _lower_ overhead by skipping tuple creation and `pushToQueue` call `if !RLS().doLog`.

Really want to be doing as little as possible when not enabled.